// modules/cp/routerCp.js
var utility = require('utlity.js');
var fs = require("fs");
var user = require("moduleUser");

exports.setProfile = function(res, req){
	var uid = res.body.uid;
	var profile = res.body.profile;
	var client = utility.prepareDb();
	var querystring = "update d set profile='" + profile + "'" + "where personid='" + uid +"'";
	client.query(querystring, function(e, r){
		if(e){
			throw e;
		}
		client.end();
		req.send('ok');
	});
};

exports.returnProfile = function(res, req){
	var uid = res.body.uid;
	var querystring = "select profile from d where personid='" + uid + "'";
	var client = utility.prepareDb();
	var retuenData = {};
	retuenData.profile = '';
	retuenData.type = '';
	client.query(querystring, function(e, r){
		if(e){
			throw e;
		}
		if(r.length == 0 || !(r[0].profile)){
			//profile字段和avatar为空的二种情况: []和[{profile:[]}]
			//profile 字段为空
			var queryString = "select avatar from d where personid='" + uid + "'";
			client.query(queryString, function(e, r){
				if(e){
					throw e;
				}
				if(r.length == 0 || !(r[0].avatar)){
					//avatar 字段为空
					client.end();
					retuenData.profile = '';
					retuenData.type = 'empty';
					req.send(retuenData);
				} else {
					var avatarArr = eval("("+ r[0].avatar+")");
					var profile = avatarArr[0];
					retuenData.profile = profile;
					retuenData.type = 'avatar';
					client.end();
					req.send(retuenData);
				}
			});
		} else{
			var profile = r[0].profile;
			retuenData.profile = profile;
			retuenData.type = 'profile';
			client.end();
			req.send(retuenData);
		}
	});

};

exports.userImages = function(res,req){
	req.render('user/userDetail/userImage.ejs');
};

exports.getAllImages = function(req,res){
	var uid = req.body.uid;
	var client = utility.prepareDb();
	var querystring = "select avatar from d where personid ='" + uid +"'";
	client.query(querystring, function(e, r){
		if(e){
			throw e;
		}
		res.send(r);
		client.end();
	});
};


exports.getProofPage = function(req,res){
    res.render("cp/submitProof.ejs",{"title":"提交证件"});
};

exports.submitProof = function(req,res){
	var proofType = req.body.proofType;
	var checkUsername = req.body.checkUsername;
	var checkFile = req.files.uploadFile.path;
	var checkFileName = req.files.uploadFile.name;
	if(proofType == 1){
		fs.exists("./uploads/proofPic/identitycard/" + checkUsername,function(exists){
			if(!exists){
				//不存在
				fs.mkdir("./uploads/proofPic/identitycard/" + checkUsername,function(e){
					if(e){
						throw e;
					}
					fs.rename(checkFile,"./uploads/proofPic/identitycard/" + checkUsername + "/" + checkFileName,function(){
						var client = utility.prepareDb();
						var queryString ="select uploaded from credits where username='" + checkUsername+"'";
						client.query(queryString,function(e,r){
						if(e){
							throw e;
						}
						if(r.length === 0){
							var querystring="insert into credits (uploaded,identitycard,username) values(1,0,'" + checkUsername + "')";
							client.query(querystring,function(e,r){
								if(e){
									throw e;
								}
								res.render("cp/submitProof.ejs",{"title":"提交证件"});
							});
						} else {
							var querystring = "update credits set identitycard = 0 where username='" + checkUsername + "'";
							client.query(querystring,function(e){
								if(e){
									throw e;
								}
								res.render("cp/submitProof.ejs",{"title":"提交证件"});
							});
						}
						});
					});
				});
			} else {
			//如果图片同名则不做处理,文件夹中不会出现同名的图片
			fs.rename(checkFile,"./uploads/proofPic/identitycard/" + checkUsername + "/" + checkFileName,function(){
				var client = utility.prepareDb();
				var queryString ="select uploaded from credits where username='" + checkUsername + "'";
				client.query(queryString, function(e,r){
					if(e){
						throw e;
					}
					if(r.length === 0){
						var querystring = "insert into credits (uploaded,identitycard,username) values(1,0,'" + checkUsername + "')";
						client.query(querystring,function(e,r){
							if(e){
								throw e;
							}
							res.render("cp/submitProof.ejs",{"title":"提交证件"});
						});
					} else {
						var querystring = "update credits set identitycard = 0 where username='" + checkUsername + "'";
						client.query(querystring,function(e){
							if(e){
								throw e;
							}
							res.render("cp/submitProof.ejs",{"title":"提交证件"});
						});
					}
				});
			});
			res.render("cp/submitProof.ejs",{"title":"提交证件"});
			}
		});
	} 
	else if (proofType==2){
		fs.exists("./uploads/proofPic/income/"+checkUsername,function(exists){
			if(!exists){
				//不存在
				fs.mkdir("./uploads/proofPic/income/"+checkUsername,function(e){
					if(e){
						throw e;
					}
					fs.rename(checkFile,"./uploads/proofPic/income/"+checkUsername+"/"+checkFileName,function(){
						var client=utility.prepareDb();
						var queryString="select uploaded from credits where username='"+checkUsername+"'";
						client.query(queryString,function(e,r){
							if(e){
								throw e;
							}
							if(r.length===0){
								var querystring="insert into credits (uploaded,income,username) values(1,0,'"+checkUsername+"')";
								client.query(querystring,function(e,r){
									if(e){
										throw e;
									}
									res.render("cp/submitProof.ejs",{"title":"提交证件"});
								});
							}
							else{
								var querystring="update credits set income=0 where username='"+checkUsername+"'";
								client.query(querystring,function(e,r){
									if(e){
										throw e
									}
									res.render("cp/submitProof.ejs",{"title":"提交证件"});
								});
							}
						});
					});
				});
			} else {
			//如果图片同名则不做处理,文件夹中不会出现同名的图片
			fs.rename(checkFile,"./uploads/proofPic/income/"+checkUsername+"/"+checkFileName,function(){
				var client=utility.prepareDb();
				var queryString="select uploaded from credits where username='"+checkUsername+"'";
				client.query(queryString,function(e,r){
					if(e){
						throw e;
					}
					if(r.length===0){
						var querystring="insert into credits (uploaded,income,username) values(1,0,'"+checkUsername+"')";
						client.query(querystring,function(e,r){
							if(e){
								throw e;
							}
							res.render("cp/submitProof.ejs",{"title":"提交证件"});
						});
					} else {
						var querystring="update credits set income=0 where username='"+checkUsername+"'";
						client.query(querystring,function(e,r){
							if(e){
								throw e
							}
							res.render("cp/submitProof.ejs",{"title":"提交证件"});
						});
					}
				});
			});
			res.render("cp/submitProof.ejs",{"title":"提交证件"});
			}
		});
	}
	else if(proofType==3){
		fs.exists("./uploads/proofPic/housing/"+checkUsername,function(exists){
			if(!exists){
				//不存在
				fs.mkdir("./uploads/proofPic/housing/"+checkUsername,function(e){
					if(e){
						throw e;
					}
					fs.rename(checkFile,"./uploads/proofPic/housing/"+checkUsername+"/"+checkFileName,function(){
						var client=utility.prepareDb();
						var queryString="select uploaded from credits where username='"+checkUsername+"'";
						client.query(queryString,function(e,r){
							if(e){
								throw e;
							}
							if(r.length===0){
								var querystring="insert into credits (uploaded,housing,username) values(1,0,'"+checkUsername+"')";
								client.query(querystring,function(e,r){
									if(e){
										throw e;
									}
									res.render("cp/submitProof.ejs",{"title":"提交证件"});
								});
							} else {
								var querystring="update credits set housing=0 where username='"+checkUsername+"'";
								client.query(querystring,function(e){
									if(e){
										throw e;
									}
									res.render("cp/submitProof.ejs",{"title":"提交证件"});
								});
							}
						});
					});
				});
			}
			else{
				//如果图片同名则不做处理,文件夹中不会出现同名的图片
				fs.rename(checkFile,"./uploads/proofPic/housing/"+checkUsername+"/"+checkFileName,function(){       
					var client=utility.prepareDb();
					var queryString="select uploaded from credits where username='"+checkUsername+"'";
					client.query(queryString,function(e,r){
						if(e){
							throw e;
						}
						if(r.length===0){
							var querystring="insert into credits (uploaded,housing,username) values(1,0,'"+checkUsername+"')";
							client.query(querystring,function(e,r){
								if(e){
									throw e;
								}
								res.render("cp/submitProof.ejs",{"title":"提交证件"});
							});
						} else {
							var querystring="update credits set housing=0 where username='"+checkUsername+"'";
							client.query(querystring,function(e){
								if(e){
									throw e;
								}
								res.render("cp/submitProof.ejs",{"title":"提交证件"});
							});
						}
					});
				});
				res.render("cp/submitProof.ejs",{"title":"提交证件"});
			}
		});
	}
	else if(proofType==4){
		fs.exists("./uploads/proofPic/gangao/"+checkUsername,function(exists){
			if(!exists){
				//不存在
				fs.mkdir("./uploads/proofPic/gangao/"+checkUsername,function(e){
					if(e){
						throw e;
					}
					fs.rename(checkFile,"./uploads/proofPic/gangao/"+checkUsername+"/"+checkFileName,function(){
						var client=utility.prepareDb();
						var queryString="select uploaded from credits where username='"+checkUsername+"'";
						client.query(queryString,function(e,r){
							if(e){
								throw e;
							}
							if(r.length===0){
								var querystring="insert into credits (uploaded,gangao,username) values(1,0,'"+checkUsername+"')";
								client.query(querystring,function(e,r){
									if(e){
										throw e;
									}
									res.render("cp/submitProof.ejs",{"title":"提交证件"});
								});
							} else {
								var querystring="update credits set gangao=0 where username='"+checkUsername+"'";
								client.query(querystring,function(e){
									if(e){
										throw e;
									}
									res.render("cp/submitProof.ejs",{"title":"提交证件"});
								});
							}
						});
					});
				});
			} else {
				//如果图片同名则不做处理,文件夹中不会出现同名的图片
				fs.rename(checkFile,"./uploads/proofPic/gangao/"+checkUsername+"/"+checkFileName,function(){
					var client=utility.prepareDb();
					var queryString="select uploaded from credits where username='"+checkUsername+"'";
					client.query(queryString,function(e,r){
						if(e){
							throw e;
						}
						if(r.length===0){
							var querystring="insert into credits (uploaded,gangao,username) values(1,0,'"+checkUsername+"')";
							client.query(querystring,function(e,r){
								if(e){
									throw e;
								}
								res.render("cp/submitProof.ejs",{"title":"提交证件"});
							});
						} else {
							var querystring="update credits set gangao=0 where username='"+checkUsername+"'";
							client.query(querystring,function(e){
								if(e){
									throw e;
								}
								res.render("cp/submitProof.ejs",{"title":"提交证件"});
							});
						}
					});
				});
				res.render("cp/submitProof.ejs",{"title":"提交证件"});
			}
		});
	}
	else if(proofType==5){
		fs.exists("./uploads/proofPic/drive/"+checkUsername,function(exists){
			if(!exists){
				//不存在
				fs.mkdir("./uploads/proofPic/drive/"+checkUsername,function(e){
					if(e){
						throw e;
					}
					fs.rename(checkFile,"./uploads/proofPic/drive/"+checkUsername+"/"+checkFileName,function(){
						var client=utility.prepareDb();
						var queryString="select uploaded from credits where username='"+checkUsername+"'";
						client.query(queryString,function(e,r){
							if(e){
								throw e;
							}
							if(r.length===0){
								var querystring="insert into credits (uploaded,drive,username) values(1,0,'"+checkUsername+"')";
								client.query(querystring,function(e,r){
									if(e){
										throw e;
									}
									res.render("cp/submitProof.ejs",{"title":"提交证件"});
								});
							} else {
								var querystring="update credits set drive = 0 where username='"+checkUsername+"'";
								client.query(querystring,function(e){
									if(e){
										throw e;
									}
									res.render("cp/submitProof.ejs",{"title":"提交证件"});
								});
							}
						});
					});
				});
			} else {
				//如果图片同名则不做处理,文件夹中不会出现同名的图片
				fs.rename(checkFile,"./uploads/proofPic/drive/"+checkUsername+"/"+checkFileName,function(){
				var client = utility.prepareDb();
				var queryString="select uploaded from credits where username='"+checkUsername+"'";
				client.query(queryString,function(e,r){
					if(e){
						throw e;
					}
					if(r.length===0){
						var querystring="insert into credits (uploaded,drive,username) values(1,0,'"+checkUsername+"')";
						client.query(querystring,function(e,r){
							if(e){
								throw e;
							}
							res.render("cp/submitProof.ejs",{"title":"提交证件"});
						});
					} else {
						var querystring="update credits set drive=0 where username='"+checkUsername+"'";
						client.query(querystring,function(e){
							if(e){
								throw e;
							}
							res.render("cp/submitProof.ejs",{"title":"提交证件"});
						});
					}
				});
			});
			res.render("cp/submitProof.ejs",{"title":"提交证件"});
			}
		});
	}
};
